kzg-bench/src/tests/zero_poly.rs

```diff
@@ -0,0 +1,292 @@
+
+                let partial = do_zero_poly_mul_partial(&idxs[..partial_size], 1, &fft_settings).unwrap();
+                partials.push(partial);
+            }
+
+            let from_tree_reduction = reduce_partials(point_count, &partials, &fft_settings).unwrap();
+            let from_direct = do_zero_poly_mul_partial(
+                &missing[..missing_count], fft_settings.get_max_width() / point_count, &fft_settings).unwrap();
+
+            for i in 0..(missing_count + 1) {
+                from_tree_reduction.get_coeff_at(i).equals(&from_direct.get_coeff_at(i));
+            }
+        }
+    }
+}
+
+/// Check that polynomial evaluation works against precomputed values
+pub fn check_test_data<TFr: Fr, TFFTSettings: FFTSettings<TFr>, TPoly: Poly<TFr>>(
+    fft_fr: &dyn Fn(&[TFr], bool, &TFFTSettings) -> Result<Vec<TFr>, String>
+) {
+    let mut expected_eval = TPoly::new(16);
+    let mut expected_poly = TPoly::new(16);
+
+    let fft_settings = TFFTSettings::new(4).unwrap();
+
+    for i in 0..16 {
+        expected_eval.set_coeff_at(i, &TFr::from_u64_arr(&EXPECTED_EVAL_U64[i]));
+        expected_poly.set_coeff_at(i, &TFr::from_u64_arr(&EXPECTED_POLY_U64[i]));
+    }
+
+    for i in 0..16 {
+        if !EXISTS[i] {
+            let tmp = expected_poly.eval(&fft_settings.get_expanded_roots_of_unity_at(i));
+            assert!(tmp.is_zero());
+        }
+    }
+
+    for i in 1..8 {
+        let tmp = expected_eval.eval(&fft_settings.get_expanded_roots_of_unity_at(i));
+        assert!(tmp.is_zero());
+    }
+
+    let tmp_poly = fft_fr(&expected_eval.get_coeffs(), true, &fft_settings).unwrap();
+
+    for i in 0..16 {
+        assert!(tmp_poly[i].equals(&expected_poly.get_coeff_at(i)));
+    }
+}
+
+/// Check if zero polynomial is calculated and evaluated as expected against precomputed values
+pub fn zero_poly_known<TFr: Fr, TFFTSettings: FFTSettings<TFr>, TPoly: Poly<TFr>>(
+    zero_poly_via_multiplication: &dyn Fn(usize, &[usize], &TFFTSettings) -> Result<(Vec<TFr>, TPoly), String>,
+) {
+    let fft_settings = TFFTSettings::new(4).unwrap();
+
+    let mut missing_idxs = Vec::new();
+    let mut expected_eval = TPoly::new(16);
+    let mut expected_poly = TPoly::new(16);
+
+    for i in 0..16 {
+        expected_eval.set_coeff_at(i, &TFr::from_u64_arr(&EXPECTED_EVAL_U64[i]));
+        expected_poly.set_coeff_at(i, &TFr::from_u64_arr(&EXPECTED_POLY_U64[i]));
+
+        if !EXISTS[i] {
+            missing_idxs.push(i);
+        }
+    }
+
+    let (zero_eval, zero_poly) =
+        zero_poly_via_multiplication(16, &missing_idxs, &fft_settings).unwrap();
+
+    for i in 0..expected_eval.len() {
+        assert!(expected_eval.get_coeff_at(i).equals(&zero_eval[i]));
+        assert!(expected_poly.get_coeff_at(i).equals(&zero_poly.get_coeff_at(i)));
+    }
+}
+
+/// Generate random series of missing indices and check if they are multiplied correctly
+pub fn zero_poly_random<TFr: Fr, TFFTSettings: FFTSettings<TFr>, TPoly: Poly<TFr>>(
```

Comment:
@Julius-Syvis @Mantulis00 ir kiti

Perdarykit normaliai šitą ir visas kitas testų funkcijas kuriose į parametrus kišate funkcijas, kurios vėliau atsiranda `kzg-bench` crate, nes galima apsieiti daug paprasčiau ir be jų: čia `FFTSettings` turėtų turėti kažkokį tai `zero_poly_mul` trait'ą, grąžinantį gatavą polinomą ir kažkokį Fr, o `fft_fr` -- Fr, nes pagal c-kzg source failus:

```C
C_KZG_RET fft_fr(fr_t *out, const fr_t *in, bool inverse, uint64_t n, const FFTSettings *fs)
```
akivaizdžiai matome, kad `FFTSettings`, kurio reikšmė nėra modifikuojama, grąžina `fr_t *out`

```C
C_KZG_RET zero_polynomial_via_multiplication(fr_t *zero_eval, poly *zero_poly, uint64_t length, const uint64_t *missing_indices, uint64_t len_missing, const FFTSettings *fs)
```
čia lygiai tas pats -- yra grąžinami `fr_t *zero_eval` ir `poly *zero_poly` panaudojant tą patį `FFTSettings`

Reply:
Taip, tu teisus, čia mūsų implementacijos bruožas. Pakeisim

Reply:
Įkėliau kątik 3 naujus traitus su šitom funkcijom, gali pažiūrėt